### PR TITLE
Added network argument to nova boot + fix id

### PR DESCRIPTION
**Pull request by ghost**
Added support to nova boot for user to specify --nic net-id='public', --nic net-id='private', --nic net-id=private_vlan_id, or --nic net-id=private_vlan_id --nic net-id=public_vlan_id

where private_vlan_id and public_vlan_id are actual vlan ids.

Also converted id to string in subnet/network.

### CODE REVIEW

**Comment by sudorandom** [comment]
+1

File: jumpgate/compute/drivers/sl/servers.py [hunk 1]
```diff
@@ -167,39 +167,65 @@ def on_get(self, req, resp, tenant_id):
...
     def on_post(self, req, resp, tenant_id):
+        payload = {}
         client = req.env['sl_client']
         body = json.loads(req.stream.read().decode())
+
+        payload['hostname'] = body['server']['name']
+        payload['domain'] = config.CONF['default_domain'] or 'jumpgate.com'
+        payload['image_id'] = body['server']['imageRef']
+
+        # TODO(kmcdonald) - How do we set this accurately?
```
Comment (jimlindeman) [inline jumpgate/compute/drivers/sl/servers.py:178]: This makes sense to allow overriding this via flavor "extra specs", now that those can be configured in jumpgate.conf

Comment (ghost) [inline jumpgate/compute/drivers/sl/servers.py:178]: Okay, I'll create a new issue for this and I will fix this on my next pull request.

**Comment by jimlindeman** [comment]
+1
Currently, it shows this pull request needs merge conflicts resolved in tests/jumpgate-tests/compute/test_servers.py
It looks like you will need to merge the changes from:
https://github.com/softlayer/jumpgate/commit/f4f365e1f85e3aa427e6c8053620766228ee3c37

**Comment by ghost** [comment]
It is now merged.

**Pull request merged**